core/src/test/java/org/springframework/security/authentication/UserDetailsRepositoryReactiveAuthenticationManagerTests.java

```diff
@@ -171,4 +172,56 @@ public void authenticateWhenPostAuthenticationChecksNotSet() {
 
 		verifyZeroInteractions(this.postAuthenticationChecks);
 	}
+
+	@Test(expected = AccountExpiredException.class)
+	public void authenticateWhenAccountExpiredThenException() {
+		this.manager.setPasswordEncoder(this.encoder);
+
+		UserDetails expiredUser = User.withUsername("user")
+				.password("password")
+				.roles("USER")
+				.accountExpired(true)
+				.build();
+		when(this.userDetailsService.findByUsername(any())).thenReturn(Mono.just(expiredUser));
+
+		UsernamePasswordAuthenticationToken token = new UsernamePasswordAuthenticationToken(
```

Comment:
Can we use `expiredUser` to create the token instead of `this.user`? It will mimic a more realistic scenario. Similarly for `lockedUser` and `disabledUser` in the other tests.